torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2752,8 +2752,6 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collectiveCoalesced(
         std::make_shared<std::vector<at::Tensor>>(inputs);
   }
 
-  at::cuda::OptionalCUDAGuard gpuGuard;
-
```

Comment:
Why is this guard being removed instead of updated to device?

Reply:
The `AutoNcclGroup` below does that